### PR TITLE
Rust: Refactor Tags

### PR DESCRIPTION
**Pull request by droogmic**
There are some generic type stuff in rust that I am not too happy with...

### CODE REVIEW

File: rust/osm2lanes/src/tags.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        let mut map = BTreeMap::new();
+        for line in s.lines() {
+            let (key, val) = line.split_once("=").ok_or("tag must be = separated")?;
+            map.insert(key.to_owned(), val.to_owned());
+        }
+        Ok(Self(map))
+    }
+}
+
+// TODO, shouldn't TagKey be passed by reference?
```
Comment (dabreegster) [inline rust/osm2lanes/src/tags.rs:86]: Yeah, that's what I'd expect. If you just switch from `Into` to [AsRef](https://doc.rust-lang.org/std/convert/trait.AsRef.html), that may work?

**Review by dabreegster — COMMENTED** [review]
Hmm, let me see if I can take this refactor a bit further... ok if I push a commit onto this PR?

File: rust/osm2lanes/src/tags.rs (hunk 1 above)
Comment (dabreegster) [reply rust/osm2lanes/src/tags.rs:86]: Hmm, not straightforward, not sure and have to get back to other work..

**Review by dabreegster — APPROVED** [review]
I'm happy with this if you are. Performance really is negligible, so I wouldn't worry about clones for now. As long as the API is easy to use, we can improve internals of Tags later

**Comment by droogmic** [comment]
> Hmm, let me see if I can take this refactor a bit further... ok if I push a commit onto this PR?

I see the update, thanks! I forgot to update that part :P

**Pull request merged**